### DISCUSSION TITLE
Better insight into the actual scheduled time for a DAG?

### DISCUSSION THREAD

**Discussion by waardd** [apache/airflow#30388] [Q&A]
I know there is a whole idea behind the scheduling of DAG's, but is there a better way to get insight into when its actual scheduled?
I find this confusing
![image](https://user-images.githubusercontent.com/19835721/229107336-781e84a9-ca3d-476a-9ac7-c5a4c2661de1.png)

I read the article here: https://towardsdatascience.com/airflow-schedule-interval-101-bbdda31cc463
With this example
![image](https://user-images.githubusercontent.com/19835721/229107722-ec70f73e-d8e1-435f-ada1-63b41f4d6f1e.png)

**Comment by potiuk** [comment]
What's confusing and what version are you at?

**Comment by waardd** [comment]
We are working with version:
Version: [v2.5.1](https://pypi.python.org/pypi/apache-airflow/2.5.1)
Git Version: .release:2.5.1+49867b660b6231c1319969217bc61917f7cf9829

And the confusion is more about the next scheduled time in the upper right corner and in the DAG overview
For exmaple this:
![image](https://user-images.githubusercontent.com/19835721/232465852-5c0f8ee3-8287-4c1d-92ed-85f2b2eebdeb.png)

next run states 2023-04-17, 02:00:00 but in fact its 2023-04-18 02:00:00
With the data interval between thoose two times.
It would be more clear to me if the next run would ectually say when the next run would be (in this case 2023-04-18 02:00:00)
Then its more clear when exactly the next run will be taking place without having to click something or count it out via the cron

**Reply by potiuk** [reply]
Could you please open an issue for that - detailing how you get that, whch timetable you are using and describe more of the circumstances you have ? Is it happening always/sometimes/ for all your dags/tasks every time or just occasionally ?

As of few versions, Airflow has multiple timetable implementaiton and what you see, depends on which timetable you have and what your expectation is (in some cases you run data-intervals, in some cases you might treat airflow dags as CRON jobs, and in some other cases the timetable might be completely custom (in which case the one who develops the timetables might provide those hints on next runs displayed).

Describing exaclty which timetables you have in your DAGS in an issue, if it does not match your expectations would be the righ way for those who look at the different timetables implemented and maybe it is simply misunderstanding of which timetables you use, vs what kind of workflows you have.

**Reply by potiuk** [reply]
BTW. The article was written loong before the multiple/custom timetables were introduced to Airlfow, so maybe you can also take a look at more fresh Airflow documentation -  https://airflow.apache.org/docs/apache-airflow/stable/authoring-and-scheduling/timetable.html#timetables , and then maybe you will be able to choose a better timetable for you?
